src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlColumnEncryptionCngProvider.Windows.cs

```diff
@@ -31,217 +33,80 @@ public class SqlColumnEncryptionCngProvider : SqlColumnEncryptionKeyStoreProvide
-
-            // Copy key path length
-            Buffer.BlockCopy(keyPathLength, 0, encryptedColumnEncryptionKey, currentIndex, keyPathLength.Length);
-            currentIndex += keyPathLength.Length;
-
-            // Copy ciphertext length
-            Buffer.BlockCopy(cipherTextLength, 0, encryptedColumnEncryptionKey, currentIndex, cipherTextLength.Length);
-            currentIndex += cipherTextLength.Length;
-
-            // Copy key path
-            Buffer.BlockCopy(masterKeyPathBytes, 0, encryptedColumnEncryptionKey, currentIndex, masterKeyPathBytes.Length);
-            currentIndex += masterKeyPathBytes.Length;
-
-            // Copy ciphertext
-            Buffer.BlockCopy(cipherText, 0, encryptedColumnEncryptionKey, currentIndex, cipherText.Length);
-            currentIndex += cipherText.Length;
-
-            // copy the signature
-            Buffer.BlockCopy(signedHash, 0, encryptedColumnEncryptionKey, currentIndex, signedHash.Length);
-
-            return encryptedColumnEncryptionKey;
+            return cekEncryptionParameters.Encrypt(columnEncryptionKey);
         }
 
         /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCngProvider.xml' path='docs/members[@name="SqlColumnEncryptionCngProvider"]/SignColumnMasterKeyMetadata/*' />
-        public override byte[] SignColumnMasterKeyMetadata(string masterKeyPath, bool allowEnclaveComputations)
+        public override byte[] SignColumnMasterKeyMetadata(string? masterKeyPath, bool allowEnclaveComputations)
         {
             throw new NotSupportedException();
         }
 
         /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCngProvider.xml' path='docs/members[@name="SqlColumnEncryptionCngProvider"]/VerifyColumnMasterKeyMetadata/*' />
-        public override bool VerifyColumnMasterKeyMetadata(string masterKeyPath, bool allowEnclaveComputations, byte[] signature)
+        public override bool VerifyColumnMasterKeyMetadata(string? masterKeyPath, bool allowEnclaveComputations, byte[]? signature)
         {
             throw new NotSupportedException();
         }
 
         /// <summary>
         /// This function validates that the encryption algorithm is RSA_OAEP and if it is not,
-        /// then throws an exception
+        /// then throws an exception.
         /// </summary>
-        /// <param name="encryptionAlgorithm">Asymmetric key encryptio algorithm</param>
-        /// <param name="isSystemOp">Indicates if ADO.NET calls or the customer calls the API</param>
-        private void ValidateEncryptionAlgorithm(string encryptionAlgorithm, bool isSystemOp)
+        /// <param name="encryptionAlgorithm">Asymmetric key encryption algorithm.</param>
+        /// <param name="isSystemOp">Indicates if ADO.NET calls or the customer calls the API.</param>
+        private static void ValidateEncryptionAlgorithm([NotNull] string? encryptionAlgorithm, bool isSystemOp)
```

Comment:
My followup question is - if we don't want null, why do we have the `string?` here? Why not make it `string` and it implicitly becomes not-null. If it might be not-null somewhere else, we can use `variableName!` to assert it's not null, right? Since it's private the scope of it should be pretty small